### PR TITLE
EN-55 create import component

### PR DESCRIPTION
**Pull request by grzegorzsz18**

### CODE REVIEW

File: src/main/java/pl/psnc/dei/ui/components/imports/CreateImportComponent.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+		add(switchingTables);
+		actionButtons = generateActionButtons();
+		add(actionButtons);
+	}
+
+	private void createComponent() {
+		if (anImport == null) {
+			add(createProjectSelection());
+			HorizontalLayout importNameLayout = new HorizontalLayout();
+			importName = new Input();
```
Comment (helinek) [inline src/main/java/pl/psnc/dei/ui/components/imports/CreateImportComponent.java]: When this is a new import I think we should give it a default name with possibility to change it.

File: src/main/java/pl/psnc/dei/ui/components/imports/CreateImportComponent.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+			statusLayout.add(new Label(anImport.getStatus().toString()));
+			add(statusLayout);
+		}
+		refresh();
+	}
+
+	private Component generateSwitchingButtons() {
+		VerticalLayout switchingButtons = new VerticalLayout();
+		switchingButtons.setMaxWidth("100px");
+		Button addToSelected = new Button("->");
```
Comment (helinek) [inline src/main/java/pl/psnc/dei/ui/components/imports/CreateImportComponent.java]: Isn't there an icon for this in Vaadin resources? I think there should be...

File: src/main/java/pl/psnc/dei/ui/components/imports/CreateImportComponent.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+		return anImport != null && !(ImportStatus.IN_PROGRESS.equals(anImport.getStatus()) && ImportStatus.SENT.equals(anImport.getStatus()));
+	}
+
+	private boolean shouldShowCreateButton() {
+		return anImport == null;
+	}
+
+	private Grid<Record> generateRecordsGrid(Set<Record> records) {
+		Grid<Record> recordsGrid = new Grid<>();
+		recordsGrid.setWidth("600px");
```
Comment (helinek) [inline src/main/java/pl/psnc/dei/ui/components/imports/CreateImportComponent.java]: Generally I would avoid putting sizes and other visual settings in code. It's better to move them to css and eventually change them later on that level.

File: src/main/java/pl/psnc/dei/ui/pages/ImportPage.java [hunk 4]
```diff
@@ -45,6 +56,30 @@ public void createListImports() {
 		displayingPlace = new ImportsListComponent(importsRepository, this);
 		add(displayingPlace);
 	}
+
+	public void createImportCreate() {
```
Comment (helinek) [inline src/main/java/pl/psnc/dei/ui/pages/ImportPage.java]: Why is this method named like that?

File: src/main/java/pl/psnc/dei/service/ImportPackageService.java [hunk 5]
```diff
@@ -49,12 +46,36 @@ private static String createImportName(String name, String projectName) {
         return StringUtil.isNullOrEmpty(name) ? generateImportName(projectName) : name;
     }
 
+    public void updateImport(Import updatedImport, Set<Record> records) {
```
[the remark this answers is not in the document]
Comment (grzegorzsz18) [reply src/main/java/pl/psnc/dei/service/ImportPackageService.java]: UpdatedImport is used below to find all records which belong to it before update.  This operators are in guava and I would have to add them to the project to use them

**Comment by krminta** [comment]
Looks fine for me.

**Pull request merged**